### PR TITLE
Added OIDC logout handler for cross protocol logout feature

### PR DESCRIPTION
**Pull request by DeshanKTD**
### Proposed changes in this pull request

OIDCLogoutHandler and OIDCLogoutHanlderTest were added.
SAML1BearerGrantHandlerTest updated for latest identity-framework.

### CODE REVIEW

**Comment by DeshanKTD** [comment]
CI build failure will be resolved after merging PR wso2/carbon-identity-framework#1525

File: pom.xml [hunk 1]
```diff
@@ -713,12 +718,12 @@
         <carbon.kernel.registry.imp.pkg.version.range>[1.0.1, 2.0.0)</carbon.kernel.registry.imp.pkg.version.range>
 
         <!-- Carbon Identity Framework version -->
-        <carbon.identity.framework.version>5.11.109</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.11.178-SNAPSHOT</carbon.identity.framework.version>
```
Comment (senthalan) [inline pom.xml]: Change to latest release version..  Don't include SNAPSHOT versions in PRs.

File: pom.xml [hunk 2]
```diff
@@ -713,12 +718,12 @@
...
         <carbon.identity.framework.imp.pkg.version.range>[5.0.0, 6.0.0)
         </carbon.identity.framework.imp.pkg.version.range>
 
         <!--SAML component version-->
-        <carbon.identity.sso.saml.version>5.3.0</carbon.identity.sso.saml.version>
+        <carbon.identity.sso.saml.version>5.4.7-SNAPSHOT</carbon.identity.sso.saml.version>
```
Comment (senthalan) [inline pom.xml]: Change to latest release version..

File: pom.xml [hunk 3]
```diff
@@ -64,6 +64,11 @@
     <dependencyManagement>
         <dependencies>
             <!-- Carbon Kernel dependencies -->
+            <dependency>
+                <groupId>com.sun.org.apache</groupId>
+                <artifactId>jaxp-ri</artifactId>
+                <version>1.4</version>
```
Comment (senthalan) [inline pom.xml]: Best practice is to add the dependencies in the parent pom under dependency management and use in the child pom. And use a property to specify the version

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML1BearerGrantHandlerTest.java [hunk 4]
```diff
@@ -203,30 +229,124 @@ public void testValidateGrant(String assertion, boolean enableAudienceRestrictio
...
+            documentBuilderFactory.setIgnoringComments(true);
+            DocumentBuilder docBuilder = documentBuilderFactory.newDocumentBuilder();
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(samlString.getBytes());
+            Document document = docBuilder.parse(inputStream);
+            Element element = document.getDocumentElement();
+            UnmarshallerFactory unmarshallerFactory = Configuration.getUnmarshallerFactory();
+            Unmarshaller unmarshaller = unmarshallerFactory.getUnmarshaller(element);
+            xmlObject = unmarshaller.unmarshall(element);
+        } catch (UnmarshallingException | SAXException | IOException | ParserConfigurationException e) {
+            String message = "Error in constructing XML Object from the encoded String";
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML1BearerGrantHandlerTest.java]: What are we doing if an exception is caught?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML1BearerGrantHandlerTest.java [hunk 5]
```diff
@@ -203,30 +229,124 @@ public void testValidateGrant(String assertion, boolean enableAudienceRestrictio
...
+
+        IdentityProviderManager identityProviderManager = mock(IdentityProviderManager.class);
+        mockStatic(IdentityProviderManager.class);
+        when(identityProviderManager.getIdPByAuthenticatorPropertyValue(anyString(), anyString(),
+                anyString(), anyBoolean())).thenReturn(identityProvider);
+        when(identityProviderManager.getResidentIdP(anyString())).thenReturn(identityProvider);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+
+        this.mockAuthenticators(identityProvider);
+
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML1BearerGrantHandlerTest.java]: Please remove the additional new line.

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/internal/OIDCSessionManagementComponent.java [hunk 6]
```diff
@@ -85,6 +87,9 @@ protected void activate(ComponentContext context) {
         if (log.isDebugEnabled()) {
             log.debug("ClaimProvider bundle is activated");
         }
+
+        context.getBundleContext().registerService(AbstractEventHandler.class.getName(),  new OIDCLogoutHandler(), null);
+
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/internal/OIDCSessionManagementComponent.java]: Remove the additional new line, please.

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 7]
```diff
@@ -70,6 +70,16 @@
...
     private static final long serialVersionUID = -9203934217770142011L;
 
+    /**
+     * Returns the OpenIDConnect User Consent.
+     *
+     * @return
+     */
+    private static boolean getOpenIDConnectSkipeUserConsent() {
+        return OAuthServerConfiguration.getInstance().getOpenIDConnectSkipeUserConsentConfig();
+
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: Move this new line to the beginning of the method.

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 8]
```diff
@@ -303,7 +315,8 @@ private String getTenantDomainForSignatureValidation(String idToken) {
 
     /**
      * Send request to consent URI
-     * @param request Http servlet request
+     *
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: New line in wrong place. Move it up, please.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/backChannelLogout/OIDCLogoutHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+        oAuthAppDOOne.setUser(authenticatedUser);
+        AppInfoCache.getInstance().addToCache(OIDCLogoutConstants.clientIdOne, oAuthAppDOOne);
+        OAuthAppDO oAuthAppDOTwo = new OAuthAppDO();
+        oAuthAppDOTwo.setBackChannelLogoutUrl(OIDCLogoutConstants.backChannelUrlTwo);
+        oAuthAppDOTwo.setUser(authenticatedUser);
+        AppInfoCache.getInstance().addToCache(OIDCLogoutConstants.clientIdTwo, oAuthAppDOTwo);
+
+        // creating mocks
+        createMocks();
+
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/backChannelLogout/OIDCLogoutHandlerTest.java]: Please remove the newline.

**Pull request merged**